tests/test_code_actions.py

```diff
@@ -257,6 +257,30 @@ def test_requests_with_no_diagnostics(self) -> Generator:
         self.assertEquals(annotations_range[0].a, 3)
         self.assertEquals(annotations_range[0].b, 0)
 
+    def test_extends_range_to_include_diagnostics(self) -> Generator:
+        def handle_response(actions_by_config: CodeActionsByConfigName) -> None:
+            pass
+        self.insert_characters('x diagnostic')
+        yield from self.await_message("textDocument/didChange")
+        yield from self.await_client_notification(
+            "textDocument/publishDiagnostics",
+            create_test_diagnostics([
+                ('diagnostic word', Range(Point(0, 2), Point(0, 12))),
+                ('all content', Range(Point(0, 0), Point(0, 12))),
+            ])
+        )
+        self.view.sel().clear()
+        self.view.sel().add(sublime.Region(0, 5))
```

Comment:
You cannot modify the cursor without an edit object. This should use `self.view.run_command("lsp_selection_set", ...)`.

Reply:
Cool stuff

Reply:
That actually fixes the problem that `on_modified_async` was not triggered and I had to trigger it manually. :)

Reply:
Done